halpybot/packages/seals/userinfo.py

```diff
@@ -19,42 +19,29 @@ async def whois(engine: Engine, subject):
         subject (str): The Seal's name being searched
 
     Returns:
-        (str): The details about the Seal, or an error string if not found.
+        (Seal): The Seal object
 
     """
-    # Set default values
-    u_id, u_cases, u_name, u_regdate, u_distant_worlds, u_distant_worlds_2 = (
-        None,
-        None,
-        None,
-        None,
-        None,
-        None,
-    )
     connection = engine.raw_connection()
-    try:
-        args = (subject, 0, 0, 0, 0, 0, 0)
-        cursor_obj = connection.cursor()
-        cursor_obj.callproc("spWhoIs", args)
-        result = list(cursor_obj.fetchall())
-        cursor_obj.close()
-        connection.commit()
-    except NoDatabaseConnection:
-        return "Error searching user."
-    finally:
-        connection.close()
+    args = (subject, 0, 0, 0, 0, 0, 0, 0)
+    cursor_obj = connection.cursor()
+    cursor_obj.callproc("spWhoIs", args)
+    result = list(cursor_obj.fetchall())
+    cursor_obj.close()
+    connection.commit()
```

Comment:
this operation appears to be read only, why is there a commit here?

Reply:
Blame ctl+c ctl+v. Guess we don't need it there after all :D

Reply:
you only need commits in order to affect change to the database.
read-only queries need not commit